src/engines/contract_engine.rs

```diff
@@ -171,21 +171,37 @@ pub async fn dry_run_call(
 mod tests {
 	use super::*;
 	use std::fs;
-	use tempdir;
+	use std::path::PathBuf;
+    use tempdir::TempDir;
+
+	fn setup_test_environment() -> Result<TempDir, Box<dyn std::error::Error>> {
+        let temp_contract_dir = TempDir::new("test_folder")?;
+        let result: anyhow::Result<()> = create_smart_contract("test".to_string(), &Some(PathBuf::from(temp_contract_dir.path())));
+        assert!(result.is_ok(), "Failed to create smart contract");
```

Comment:
nitpick: I know this is a test, but this function returns a result, so maybe make use of the `Err` on it? `anyhow::bail!`?

Reply:
```rust
if !result.is_ok() {
	anyhow::bail!("Failed to create smart contract");
}
```
Something like this?

Reply:
Yep.